crates/rolldown_error/src/events/ambiguous_external_namespace.rs

```diff
@@ -0,0 +1,46 @@
+use super::BuildEvent;
+use arcstr::ArcStr;
+use oxc::span::Span;
+
+use crate::{
+  build_error::severity::Severity, diagnostic::Diagnostic,
+  types::diagnostic_options::DiagnosticOptions, EventKind,
+};
+
+#[derive(Debug)]
+pub struct AmbiguousExternalNamespace {
+  pub importer: String,
+  pub importee: Vec<String>,
+  pub importer_source: ArcStr,
+  pub importer_filename: String,
+  pub imported_specifier: String,
+  pub imported_specifier_span: Span,
+}
+
+impl BuildEvent for AmbiguousExternalNamespace {
+  fn kind(&self) -> EventKind {
+    EventKind::AmbiguousExternalNamespace
+  }
+
+  fn message(&self, _opts: &DiagnosticOptions) -> String {
+    format!(
+      r#""{}" re-exports "{}" from one of the modules {} (will be ignored)."#,
+      self.importer,
+      self.imported_specifier,
+      self.importee.iter().map(|v| format!(r#""{v}""#)).collect::<Vec<_>>().join(" and ")
+    )
+  }
+
+  fn on_diagnostic(&self, diagnostic: &mut Diagnostic, opts: &DiagnosticOptions) {
+    let file_id = diagnostic.add_file(self.importer_filename.clone(), self.importer_source.clone());
+
+    diagnostic.title = "Found ambiguous export.".to_string();
+    diagnostic.severity = Severity::Warning;
```

Comment:
Don't need to specify severity here.

See

https://github.com/rolldown/rolldown/blob/65b119ab4c42584735738288a6c3705b16e854e1/crates/rolldown/src/ast_scanner/mod.rs#L473-L481